eng/Update-PkgVersion.ps1

```diff
@@ -0,0 +1,142 @@
+<#
+.SYNOPSIS
+Bumps up package versions after release
+
+.DESCRIPTION
+This script bumps up package versions following conventions defined at https://github.com/Azure/azure-sdk/blob/master/docs/policies/releases.md#incrementing-after-release-cpp
+
+.PARAMETER RepoRoot
+The Root of the repo
+
+.PARAMETER ServiceDirectory
+The Name of the Service Directory
+
+.PARAMETER PackageName
+The Name of the Package
+
+.PARAMETER PackageDirName
+Used in the case where the package directory name is different from the package name. e.g in cognitiveservice packages
+
+.PARAMETER NewVersionString
+Use this to overide version incement logic and set a version specified by this parameter
+
+
+.EXAMPLE
+Updating package version for Azure.Core
+Update-PkgVersion.ps1 -ServiceDirectory core -PackageName Azure.Core
+
+Updating package version for Azure.Core with a specified verion
+Update-PkgVersion.ps1 -ServiceDirectory core -PackageName Azure.Core -NewVersionString 2.0.5
+
+Updating package version for Microsoft.Azure.CognitiveServices.AnomalyDetector
+Update-PkgVersion.ps1 -ServiceDirectory cognitiveservices -PackageName Microsoft.Azure.CognitiveServices.AnomalyDetector -PackageDirName AnomalyDetector
+
+#>
+
+[CmdletBinding()]
+Param (
+    [ValidateNotNullOrEmpty()]
+    [string] $RepoRoot = "${PSScriptRoot}/..",
+    [Parameter(Mandatory=$True)]
+    [string] $ServiceDirectory,
+    [Parameter(Mandatory=$True)]
+    [string] $PackageName,
+    [string] $PackageDirName,
+    [string] $NewVersionString
+)
+# Regular expression as specified in https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+$VERSION_REGEX = "^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"
+
+# Updated Version in csproj and changelog using computed or set NewVersionString
+function Update-Version($Unreleased=$True, $ReplaceVersion=$False)
+{
+    Write-Verbose "New Version: ${NewPackageVersion}"
+    Set-Content -Path $PackageVersionPath -Value ${NewPackageVersion}
+
+    # Increment Version in ChangeLog file
+    & "${PSScriptRoot}/common/Update-Change-Log.ps1" -Version $NewPackageVersion -ChangeLogPath $ChangelogPath -Unreleased $Unreleased -ReplaceVersion $ReplaceVersion
+}
+
+# Parse a VersionString to verify that it is right
+function Parse-Version($VerisionString)
+{
+    if ([System.String]::IsNullOrEmpty($VerisionString))
```

Comment:
https://github.com/Azure/azure-sdk/blob/6903dad6131be2a81469ae2d4a55f408469ad1ee/eng/scripts/Update-Release-Versions.ps1#L26-L58

That is some code we have that we are using to parse versions. We should probably figure out a way to share that but at the very least we should be consistent with it.

Reply:
Looks like you just copied it from .NET (https://github.com/Azure/azure-sdk-for-net/blob/master/eng/Update-PkgVersion.ps1#L48)... even more reason to refactor this. @chidozieononiwu can you create an issue to create some shared versioning helper scripts.

Reply:
Created https://github.com/Azure/azure-sdk-tools/issues/469 to track